### ISSUE TITLE
Improving synergy between DataFrames and scikit-learn (ML) tools

### ISSUE THREAD

**Issue by JovanVeljanoski** [python-sprints/dataframe-summit#9]
There is already a decent synergy between `pandas` and `scikit-learn` and most other popular machine learning libraries, as in a `pandas` DataFrame is almost always accepted as an input data structure. 

However, the output of the `scikit-learn` transformers is a pure `numpy` array, and thus one loses the column name information of the input data. Preserving the column names through the ML pipeline would be extremely useful to data scientists to optimize/understand/debug data science pipelines.

**Comment by toobaz** [comment]
I agree. It would also be nice if, when passed ``Series`` and ``DataFrames``, ``scikit-learn`` ML methods returned ``Series`` (or ``DataFrames`` when there are multiple labels) with corresponding indexes when predicting.
